java/bundles/org.eclipse.set.feature.plazmodel/src/org/eclipse/set/feature/plazmodel/check/GeoCoordinateValid.java

```diff
@@ -0,0 +1,167 @@
+	@Reference
+	GeoKanteGeometryService geometryService;
+
+	@Reference
+	EventAdmin eventAdmin;
+
+	@Override
+	public void handleEvent(final Event event) {
+		final Map<String, Class<? extends PlazCheck>> properties = new HashMap<>();
+		properties.put("org.eclipse.e4.data", this.getClass()); // $NON-NLS-1$
+		eventAdmin.sendEvent(new Event(Events.DO_PLAZ_CHECK, properties));
+	}
+
+	@Override
+	protected List<PlazError> run(
+			final MultiContainer_AttributeGroup container) {
+		final List<PlazError> result = new ArrayList<>();
+		final List<Punkt_Objekt> punktObjekts = StreamSupport
+				.stream(container.getAllContents().spliterator(), false)
+				.filter(Punkt_Objekt.class::isInstance)
+				.map(Punkt_Objekt.class::cast).toList();
+		punktObjekts.forEach(po -> po.getPunktObjektTOPKante().forEach(potk -> {
+			final GEOKanteCoordinate geoKanteCoordinate = geometryService
+					.getCoordinateAt(potk, 0);
+			final Coordinate coordinate = geoKanteCoordinate.getCoordinate()
+					.getCoordinate();
+			final List<GEO_Punkt> givenGEOPunkts = potk.getIDGEOPunktBerechnet()
+					.stream().map(e -> e.getValue())
+					.filter(gp -> getNullableObject(gp,
+							e -> e.getGEOPunktAllg().getGEOKoordinatensystem()
+									.getWert()).orElse(null).equals(
+											geoKanteCoordinate.getCRS()))
+					.toList();
+			givenGEOPunkts.forEach(gp -> {
+				final Coordinate gpCoordinate = GeoPunktExtensions
+						.getCoordinate(gp);
+				if (gpCoordinate == null) {
+					result.add(creatErrorReport(gp));
+				} else if (coordinate.distance(gpCoordinate) > TOLERANT) {
+					result.add(createErrorReport(po, potk, gp));
+				}
+			});
+		})
+
+		);
+		return result;
+	}
+
+	private PlazError creatErrorReport(final GEO_Punkt gp) {
+		final PlazError plazError = PlazFactory.eINSTANCE.createPlazError();
+		plazError.setSeverity(ValidationSeverity.ERROR);
+		EObject errObj = gp;
+		if (gp.getGEOPunktAllg() == null) {
+			errObj = gp.getGEOPunktAllg();
+		} else if (gp.getGEOPunktAllg().getGKX() == null) {
+			errObj = gp.getGEOPunktAllg().getGKX();
+		} else if (gp.getGEOPunktAllg().getGKY() == null) {
+			errObj = gp.getGEOPunktAllg().getGKY();
+		}
+		plazError.setObject(errObj);
+		plazError.setMessage(String.format("GEO_Punkt: %s fehlt Koordinate",
+				gp.getIdentitaet().getWert()));
+		plazError.setType(checkType());
+		return plazError;
+	}
+
+	private PlazError createErrorReport(final Punkt_Objekt po,
+			final Punkt_Objekt_TOP_Kante_AttributeGroup potk,
+			final GEO_Punkt gp) {
+		final PlazError plazError = PlazFactory.eINSTANCE.createPlazError();
+		plazError.setSeverity(ValidationSeverity.ERROR);
+		final ID_GEO_Punkt_ohne_Proxy_TypeClass errorObject = potk
+				.getIDGEOPunktBerechnet().stream()
+				.filter(idgp -> idgp.getValue().equals(gp)).findFirst()
+				.orElse(null);
+		if (errorObject == null) {
+			throw new IllegalArgumentException(String.format(
+					"GEO_Punkt: %s ist nicht zu Punkt_Objekt: %s gehören",
```

Comment:
@TruongQuangSB This message should be

```suggestion
					"GEO_Punkt: %s gehört nicht zu Punkt_Objekt: %s",
```